### PR TITLE
Update images, bug fixes, clean up code

### PR DESCRIPTION
**Pull request by carolynwang**
@Jeffwan @mbaijal @gautamkmr @kalyc

**Summary**

- Update images
- Bug fixes in utils functions, remove print transformation job results (it was specific to kmeans)
- Remove unused imports
- Minor fixes in mini image classification pipeline parameters

**Testing**

- Sample pipelines compile and run using default inputs

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1778)

### CODE REVIEW

File: components/aws/sagemaker/batch_transform/src/batch_transform.py [hunk 1]
```diff
@@ -56,10 +51,9 @@ def main(argv=None):
   batch_job_name = _utils.create_transform_job(client, vars(args))
   logging.info('Batch Job request submitted. Waiting for completion...')
   _utils.wait_for_transform_job(client, batch_job_name)
-  _utils.print_tranformation_job_result(args.output_location)
 
-  Path(args.output_location_file).parent.mkdir(parents=True, exist_ok=True)
-  Path(args.output_location_file).write_text(unicode(args.output_location))
+  with open('/tmp/output_location.txt', 'w') as f:
```
Comment (Ark-kun) [inline components/aws/sagemaker/batch_transform/src/batch_transform.py]: The original lines might be better for future-proof components. We've already got some bug reports about issues stemming from hard-coded paths causing problems for certain on-prem scenarios such as non-Docker OCI clusters.

File: components/aws/sagemaker/batch_transform/component.yaml [hunk 2]
```diff
@@ -71,7 +71,7 @@ outputs:
   - {name: output_location,    description: 'S3 URI of the transform job results.'}
 implementation:
   container:
-    image: carowang/kubeflow-pipeline-aws-sm:20190801-01
+    image: carowang/kubeflow-pipeline-aws-sm:latest
```
Comment (Ark-kun) [inline components/aws/sagemaker/batch_transform/component.yaml]: Using `latest` can cause existing pipelines to break once an incompatible image version is pushed (e.g. with some command-line parameter deprecated). Pinning the image version is preferable.

Comment (carolynwang) [reply components/aws/sagemaker/batch_transform/component.yaml]: Ah makes sense. Will revert back to using the image version.

File: samples/contrib/aws-samples/ground_truth_pipeline_demo/mini-image-classification-pipeline.py [hunk 3]
```diff
@@ -22,13 +22,13 @@ def ground_truth_test(region='us-west-2',
     ground_truth_train_job_name='mini-image-classification-demo-train',
     ground_truth_validation_job_name='mini-image-classification-demo-validation',
     ground_truth_label_attribute_name='category',
-    ground_truth_train_manifest_location='s3://your-bucket-name/gt-demo-images/ground-truth-demo/train.manifest',
-    ground_truth_validation_manifest_location='s3://your-bucket-name/gt-demo-images/ground-truth-demo/validation.manifest',
-    ground_truth_output_location='s3://your-bucket-name/gt-demo-images/ground-truth-demo/output',
+    ground_truth_train_manifest_location='s3://your-bucket-name/mini-image-classification/ground-truth-demo/train.manifest',
```
Comment (Jeffwan) [inline samples/contrib/aws-samples/ground_truth_pipeline_demo/mini-image-classification-pipeline.py]: Just like to confirm it's mini - small image?

Comment (carolynwang) [reply samples/contrib/aws-samples/ground_truth_pipeline_demo/mini-image-classification-pipeline.py]: Just a very small set of images

File: components/aws/sagemaker/batch_transform/src/batch_transform.py (hunk 1 above)
Comment (carolynwang) [reply components/aws/sagemaker/batch_transform/src/batch_transform.py]: Got it, went back to the original way

**Comment by Jeffwan** [comment]
/hold

Hold the changes since @carolynwang will have a bug bash for all the components today.

**Pull request merged**